addons/binding/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/handler/GardenaAccountHandler.java

```diff
@@ -178,21 +179,23 @@ public void handleCommand(ChannelUID channelUID, Command command) {
      */
     @Override
     public void onDeviceUpdated(Device device) {
-        Thing gardenaThing = getThingByUID(UidUtils.generateThingUID(device, getThing()));
-        try {
-            if (gardenaThing != null) {
-                GardenaThingHandler gardenaThingHandler = (GardenaThingHandler) gardenaThing.getHandler();
-                gardenaThingHandler.updateProperties(device);
-                for (Channel channel : gardenaThing.getChannels()) {
-                    gardenaThingHandler.updateChannel(channel.getUID());
+        for (ThingUID thingUID : UidUtils.getThingUIDs(device, getThing())) {
+            Thing gardenaThing = getThingByUID(thingUID);
+            try {
+                if (gardenaThing != null) {
+                    GardenaThingHandler gardenaThingHandler = (GardenaThingHandler) gardenaThing.getHandler();
+                    gardenaThingHandler.updateProperties(device);
+                    for (Channel channel : gardenaThing.getChannels()) {
+                        gardenaThingHandler.updateChannel(channel.getUID());
+                    }
+                    gardenaThingHandler.updateStatus(device);
                 }
-                gardenaThingHandler.updateStatus(device);
+            } catch (GardenaException ex) {
+                logger.error("There is something wrong with your thing, please recreate the thing {}",
+                        gardenaThing.getUID(), ex);
+            } catch (AccountHandlerNotAvailableException ex) {
```

Comment:
Optionally name the `ex` `ignore` and maybe change the comment to include the reason why you ignored it?